network-config-analyzer/PeerContainer.py

```diff
@@ -300,6 +300,35 @@ def _add_hep_from_yaml(self, hep_object):
 
         self._add_peer(hep)
 
+    def _add_networkset_from_yaml(self, networkset_object):
+        """
+        Add a Calico NetworkSet to the container based on the given resource instance
+        :param dict networkset_object: The networkSet object to add
+        :return: None
+        """
+        kind = networkset_object.get('kind')
+        is_global = kind == 'GlobalNetworkSet'
+        metadata = networkset_object.get('metadata', {})
+        spec = networkset_object.get('spec', {})
+        name = metadata.get('name', '')
+        if name == '':
+            print('NetworkSet must have a name', file=stderr)
```

Comment:
I think if a NetworkSet has no name we better exit the function here without creating the Peer.

Reply:
committed a fix